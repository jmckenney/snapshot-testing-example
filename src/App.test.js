import { render, screen } from '@testing-library/react';
import App from './App';

it('renders the component', () => {
  const component = render(<App />);
  // console.log(component.debug());
  expect(component.container).toMatchSnapshot();
});

describe('when component is provided a title', () => {
  it('shows the correct title', () => {
    const { container } = render(<App title="hello world"/>);
    expect(container).toMatchSnapshot();
  })
})

describe('when component has an error', () => {
  it('shows the error', () => {
    const { container } = render(<App showError={true} />);
    expect(container).toMatchSnapshot();
  })
})

describe('when component has children', () => {
  it('shows the children', () => {
    const { container } = render(<App><p>Hello children</p></App>);
    expect(container).toMatchSnapshot();
  })
})
