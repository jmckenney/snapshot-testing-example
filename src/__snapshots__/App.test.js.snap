// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1 />
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;

exports[`when component has an error shows the error 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1 />
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;

exports[`when component has children shows the children 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1 />
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
    <p>
      Hello children
    </p>
  </div>
</div>
`;

exports[`when component is provided a title has the correct title rendered 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        hello world
      </h1>
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;

exports[`when component is provided a title shows the correct title 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        hello world
      </h1>
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;
